src/Symfony/Component/Ldap/Security/LdapUserProvider.php

```diff
@@ -122,9 +123,15 @@ public function refreshUser(UserInterface $user)
 
     /**
      * {@inheritdoc}
+     *
+     * @final
      */
     public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
     {
+        if (!$user instanceof PasswordAuthenticatedUserInterface) {
```

Comment:
This looks useless to me, as only `LdapUser` is supported anyway, and it implements the interface